src/biokbase/narrative/tests/test_jobcomm.py

```diff
@@ -625,6 +687,80 @@ def mock_check_jobs(params):
             msg,
         )
 
+    @mock.patch(CLIENTS, get_mock_client)
+    def test_get_job_states__err(self):
+        """
+        """
+        # what FE would say was the last time the jobs were checked
+        NOW = time.time_ns()
+
+        # mix of terminal and not terminal
+        not_updated_ids = [JOB_COMPLETED, JOB_ERROR, JOB_TERMINATED, JOB_CREATED, JOB_RUNNING]
+        # not terminal
+        updated_ids = [BATCH_PARENT, BATCH_RETRY_RUNNING]
+
+        # error ids
+        not_found_ids = [JOB_NOT_FOUND]
+
+        job_ids = not_updated_ids + updated_ids
+        active_ids = list(set(job_ids) & set(ACTIVE_JOBS))
+
+        # output_states will be partitioned as
+        not_found_ids
+        terminal_ids = list(set(job_ids) - set(ACTIVE_JOBS))
+        not_updated_active_ids = list(set(not_updated_ids) & set(active_ids))
+        updated_active_ids = list(set(updated_ids) & set(active_ids))  # (yes, redundant)
```

Comment:
So the whole point of this test is to check that the mechanism for replacing not updated jobs with error stubs works. The most important part is to distinguish between `terminal_ids` and `not_updated_active_ids`, which should be given error stubs, from `updated_active_ids` which should send back actual output states